### PR TITLE
New `python manage.py db extract` command: imports topics from fetched j...

### PR DESCRIPTION
**Pull request by catherinedevlin**
...son

### CODE REVIEW

**Comment by DavidEBest** [comment]
If I run the following set up commands:
```
dropdb sbirezdev
createdb sbirezdev
python manage.py db upgrade
python manage.py db extract
```

I get:
```
psql:data/topics.sql:5099: ERROR:  duplicate key value violates unique constraint "references_reference_key"
DETAIL:  Key (reference)=(2.  Dolvin, D. (2008) Hypersonic International Flight Research and Experimentation (HIFIRE) Fundamental Science and Technology Development Strategy, 15th AIAA International Space Planes and Hypersonic Systems and Technologies Conference, AIAA-2008-2581.) already exists.
```

If I run it again, I get a bunch of errors like this:
```
psql:data/topics.sql:6: ERROR:  update or delete on table "topics" violates foreign key constraint "phases_topic_id_fkey" on table "phases"
DETAIL:  Key (id)=(64) is still referenced from table "phases".
DELETE 10
psql:data/topics.sql:8: ERROR:  update or delete on table "programs" violates foreign key constraint "topics_program_id_fkey" on table "topics"
DETAIL:  Key (id)=(1) is still referenced from table "topics".
DELETE 1010
psql:data/topics.sql:12: ERROR:  schema "import" already exists
SET
psql:data/topics.sql:29: ERROR:  relation "topics" already exists
psql:data/topics.sql:36: ERROR:  relation "phases" already exists
psql:data/topics.sql:43: ERROR:  relation "participating_components" already exists
psql:data/topics.sql:50: ERROR:  relation "areas" already exists
psql:data/topics.sql:57: ERROR:  relation "keywords" already exists
psql:data/topics.sql:64: ERROR:  relation "_references" already exists
psql:data/topics.sql:70: ERROR:  duplicate key value violates unique constraint "topics_pkey"
DETAIL:  Key (topics_id)=(1) already exists.
psql:data/topics.sql:75: ERROR:  duplicate key value violates unique constraint "topics_pkey"
DETAIL:  Key (topics_id)=(2) already exists.
```

How and when should I run this and am I doing something wrong?

**Comment by catherinedevlin** [comment]
Thanks for catching that - nasty subtle bug; the migrations/versions file to drop the unwanted unique constraint on `references` wasn't checked into version control.  So everything worked on mine, where the unique constraint was absent, but without that file yours had the constraint which caused the violation.

I need to clean up my .gitignore so that I notice untracked files like this; right now my untracked list is too long.

File: data/empty_topic_data.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+DELETE FROM public.participatingcomponentstopics;
+DELETE FROM public.participatingcomponents;
+DELETE FROM public.topicskeywords;
+DELETE FROM public.topicsareas;
+DELETE FROM public.references;
+DELETE FROM public.topics;
```
Comment (DavidEBest) [inline data/empty_topic_data.sql]: Phases should be in this list.

**Comment by DavidEBest** [comment]
This still doesn't _quite_ work for me on runs after the first. Is it supposed to?

I don't think the `import` namespace is being cleared out with the extract script.

**Pull request merged**